packages/react-components/react-tree/src/components/TreeItemLayout/TreeItemLayout.types.ts

```diff
@@ -15,6 +11,30 @@ export type TreeItemLayoutSlots = {
    * Icon slot that renders right after main content
    */
   iconAfter?: Slot<'div'>;
+  /**
+   * Expand icon slot,
+   * by default renders a chevron icon to indicate opening and closing
+   */
+  expandIcon?: Slot<'div'>;
+  aside?: Slot<'div'>;
+  /**
+   * actionable elements are normally buttons, menus, or other focusable elements.
+   * Those elements are only visibly available if the given tree item is currently active.
+   */
+  actions?: Slot<
+    ExtractSlotProps<
+      Slot<'div'> & {
+        /**
+         * Forces visibility of the aside/action content
+         */
+        visible?: boolean;
+      }
+    >
+  >;
+};
+
+export type TreeItemLayoutInternalSlots = TreeItemLayoutSlots & {
+  actions?: Slot<'div'>;
```

Comment:
This is required to ensure we're properly removing the `visible` property, that is available by the external props, from the actual internal slot (that should not have that property propagated to the DOM)